src/docs/release/breaking-changes/clip-behavior.md

```diff
@@ -0,0 +1,133 @@
+
+A `saveLayer` is especially expensive in older devices because it creates an offscreen render target, and a render target switch can sometimes cost about 1ms.
+
+Even without a `saveLayer`, a clip is still expensive because it applies to all subsequent draws until it’s restored. Thus a single clip may slow down the performance on hundreds of draw operations.
+
+In addition to performance issues, Flutter also suffered from some correctness issues as the clip was not managed and implemented in a single place. In several places, the `saveLayer` was inserted to the wrong place and thus it only increased the performance cost without fixing any bleeding edge artifacts.
+
+Therefore, we unified the `clipBehavior` control and its implementation in this breaking change. The default `clipBehavior` is `Clip.none` for most widgets to save performance except
+-   `ClipPath` defaults to `Clip.antiAlias`
+-   `ClipRRect` defaults to `Clip.antiAlias`
+-   `ClipRect` defaults to `Clip.hardEdge`
+-   `Stack` defaults to `Clip.hardEdge`
+-   `EditableText` defaults to `Clip.hardEdge`
+-   `ListWheelScrollView` defaults to `Clip.hardEdge`
+-   `SingleChildScrollView` defaults to `Clip.hardEdge`
+-   `NestedScrollView` defaults to `Clip.hardEdge`
+-   `ShrinkWrappingViewport` defaults to `Clip.hardEdge`
+
+## Migration guide
+
+To migrate the code, you have 4 choices:
+
+1.  Leave your code as is if your content does not need to be clipped (i.e., none of the widgets’ children expand outside their parent’s boundary). This will probably have a positive impact on your overall performance.
+2.  Add `clipBehavior: Clip.hardEdge` if you need clipping, and clipping without anti-alias is good enough for your (and your clients’) eyes. This will be the common case when you clip rectangles or shapes with very small curved areas (such as the corners of rounded rectangles).
+3.  Add `clipBehavior: Clip.antiAlias` if you need anti-aliased clipping. This will give you smoother edges at a slightly higher cost. This will be the common case when dealing with circles and arcs.
+4.  Add `clip.antiAliasWithSaveLayer` if you want the exact same behavior as before (May 2018). Be aware that it’s very costly in performance. This will be only rarely needed. One case where you might need this is if you have an image overlaid on a very different background color. In these cases, consider whether you can avoid overlapping multiple colors in one spot (e.g. by having the background color only present where the image is absent).
+
+For Stack widget specifically, if you previously used `overflow: Overflow.visible`, you shall simply remove `overflow: Overflow.visible` as now we default to not clip. If you used `overflow: Overflow.clip` or relied on the default `overflow: Overflow.clip` setup, you can now replace `overflow: Overflow.clip` with `clipBehavior: Clip.hardEdge`.
```

Comment:
I thought we decided to default the stack to hardEdges?

Reply:
Yes, thanks! Updated.